test/units/utils/test_encrypt.py

```diff
@@ -249,3 +249,55 @@ def test_passlib_bcrypt_salt(recwarn):
 
     result = passlib_obj.hash(secret, salt=repaired_salt, ident=ident)
     assert result == expected
+
+
+def test_do_encrypt_no_lib(mocker):
+    """Test AnsibleError is raised when no encryption library is installed."""
+    mocker.patch('ansible.utils.encrypt.HAS_CRYPT', False)
+    mocker.patch('ansible.utils.encrypt.PASSLIB_AVAILABLE', False)
+    with pytest.raises(AnsibleError, match=r"Unable to encrypt nor hash, either libxcrypt \(recommended\), crypt, or passlib must be installed."):
+        encrypt.do_encrypt("123", "sha256_crypt", salt="12345678")
+
+
+class TestCryptHash:
+    """Tests for the CryptHash class.
+
+    These tests are hitting code paths that are otherwise impossible to reach
+    through integration tests, but necessary for more complete code coverage.
+    """
+
+    def test_invalid_instantiation(self, mocker):
+        """Should not be able to instantiate a CryptHash class without libxcrypt/libcrypt."""
+        mocker.patch('ansible.utils.encrypt.HAS_CRYPT', False)
+        with pytest.raises(AnsibleError, match=r"crypt cannot be used as the 'libxcrypt' library is not installed or is unusable."):
```

Comment:
`.` is any char in regex
```suggestion
        with pytest.raises(AnsibleError, match=r"crypt cannot be used as the 'libxcrypt' library is not installed or is unusable\."):
```